roles/tftp-server/tasks/pxe.yml

```diff
@@ -0,0 +1,103 @@
+---
+- name: install the tftp server
+  apt:
+    name: dnsmasq
+
+- name: enable tftp
+  copy:
+    src: files/tftp.conf
+    dest: /etc/dnsmasq.d
+
+- name: create boot image directories
+  file:
+    path: /srv/tftp/{{item}}
+    state: directory
+    recurse: true
+    mode: 0755
+  with_items:
+    - ipxe
+    - util/cz/exper/live
+
+- name: magicly build /srv/tftp/boot/grub
+  command: grub-mknetdir
+
+- name: write grub.cfg
+  template:
+    src: grub.cfg.j2
+    dest: /srv/tftp/boot/grub/grub.cfg
+
+- name: create debian suite dirs
+  file:
+    path: /srv/tftp/debian/{{ item[0] }}/{{ item[1] }}
+    state: directory
+    recurse: true
+    mode: 0755
+  with_nested:
+    - "{{ debian_suites }}"
+    - "{{ archs }}"
+
+
+- name: create utunbu suite dirs
+  file:
+      # path: /srv/tftp/ubuntu/{{suite}}/{{arch}}
+    path: /srv/tftp/ubuntu/{{ item[0] }}/{{ item[1] }}
+    state: directory
+    recurse: true
+    mode: 0755
+  with_nested:
+    - "{{ ubuntu_suites }}"
+    - "{{ archs }}"
+
+- name: download debian netboot images
+  get_url:
+    url: http://ftp.debian.org/debian/dists/{{item[0]}}/main/installer-{{item[1]}}/current/images/netboot/debian-installer/{{item[1]}}/{{item[2]}}
+    dest: /srv/tftp/debian/{{ item[0] }}/{{ item[1] }}/{{ item[2] }}
+  with_nested:
+    - "{{ debian_suites }}"
+    - "{{ archs }}"
+    - [ 'linux', 'initrd.gz' ]
+
+- name: download ubuntu netboot images
+  get_url:
+    url: http://{{ubuntu_host}}/ubuntu/dists/{{item[0]}}/main/installer-{{item[1]}}/current/images/netboot/ubuntu-installer/{{item[1]}}/{{item[2]}}
+    dest: /srv/tftp/ubuntu/{{ item[0] }}/{{ item[1] }}/{{ item[2] }}
+  with_nested:
+    - "{{ ubuntu_suites }}"
+    - "{{ archs }}"
+    - [ 'linux', 'initrd.gz' ]
+
+- name: download ubuntu updated netboot images
```

Comment:
Why do we need both? Surely the release pocket is fine?

Reply:
yeah, this makes me a little grumpy.

(10:38:53 AM) CarlFK: why is there installer-amd64 in http://archive.ubuntu.com/ubuntu/dists/artful/main/ but not http://archive.ubuntu.com/ubuntu/dists/artful-updates/main/
(10:40:00 AM) CarlFK: I have a script that will wget http://archive.ubuntu.com/ubuntu/dists/artful/main/installer-amd64/current/images/hd-media/boot.img.gz
(10:40:14 AM) CarlFK: except from -updates, and so it breaks.
(11:25:49 AM) cjwatson: Because the installer hasn't been published to artful-updates yet
(11:26:07 AM) cjwatson: And won't be until/unless it's SRUed post-release
(11:26:24 AM) cjwatson: You'll have to try -updates and then fall back to the bare series

Reply:
I'd just stick to bare series unless there's a good reason not to. (e.g. bugs)

Reply:
need the current version to avoid kernel / module mismatch.